pkg/apiserver/registry/stats/nodelatencystats/rest_test.go

```diff
@@ -36,24 +36,63 @@ func TestREST(t *testing.T) {
 }
 
 func TestRESTCreate(t *testing.T) {
-	summary := &statsv1alpha1.NodeLatencyStats{
-		ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
-		PeerNodeLatencyStats: nil,
-	}
-	expectedObj := &statsv1alpha1.NodeLatencyStats{
-		ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
-		PeerNodeLatencyStats: nil,
+	addedTime := metav1.Now()
+	tests := []struct {
+		name        string
+		summary     *statsv1alpha1.NodeLatencyStats
+		nodeName    string
+		expectedObj runtime.Object
+		err         error
+		timeAdded   bool
+	}{
+		{
+			name: "create with time added",
+			summary: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1", CreationTimestamp: addedTime},
+				PeerNodeLatencyStats: nil,
+			},
+			nodeName: "node1",
+			expectedObj: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1", CreationTimestamp: addedTime},
+				PeerNodeLatencyStats: nil,
+			},
+			err:       nil,
+			timeAdded: true,
+		},
+		{
+			name: "create with time not added",
+			summary: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1"},
+				PeerNodeLatencyStats: nil,
+			},
+			nodeName: "node1",
+			expectedObj: &statsv1alpha1.NodeLatencyStats{
+				ObjectMeta:           metav1.ObjectMeta{Name: "node1", CreationTimestamp: metav1.Time{}},
+				PeerNodeLatencyStats: nil,
+			},
+			err:       nil,
+			timeAdded: false,
+		},
 	}
 
-	r := NewREST()
-	ctx := context.Background()
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := NewREST()
+			ctx := context.Background()
+			obj, err := r.Create(ctx, tt.summary, nil, nil)
+			require.NoError(t, err)
 
-	obj, err := r.Create(ctx, summary, nil, nil)
-	require.NoError(t, err)
-	assert.Equal(t, expectedObj, obj)
+			if tt.timeAdded {
+				assert.Equal(t, tt.expectedObj, obj)
+			} else {
+				assert.NotEqual(t, tt.expectedObj, obj)
+			}
```

Comment:
this doesn't make sense IMO. Just because a timestamp is added by `Create` in this case doesn't mean that `assert.NotEqual` is the right thing to check. Especially when you have a field called `tt.expectedObj`, which implies that you are expecting the function to return a specific object.

usually the best way to deal with this is to add a `clock.Clock` (https://pkg.go.dev/k8s.io/utils/clock#Clock) to the object being tested (in this case, to the `REST` object). In the "normal" case, the clock is set to `RealClock`, but for tests you can set it to `FakeClock`. This way you can freeze time and predict what the timestamp will be / should be in the returned object. There are multiple examples in the Antrea code base, so you can look for them.